Support for Rails 5.1

britth opened the pull request:
* `ActiveSupport::Callbacks::Filters::Environment` no longer supports
[run_block](https://github.com/rails/rails/blob/5-1-stable/activesupport/lib/active_support/callbacks.rb#L158); remove in propagate_callbacks_to_embedded_associations
* `ActiveSupport::Callbacks::CallbackSequence` [call method](https://github.com/rails/rails/commit/871ca21f6a1d65c0ec78cb5a9641411e2210460b#diff-1ecd313ff0ab827af30014553cf8918dL453) no longer
supported; update logic in propagate_callbacks_to_embedded_associations
to call invoke_before on the sequence for :before callbacks and
invoke_after for :after callbacks
* Need to explicitly [throw :abort to halt callback chain](https://blog.bigbinary.com/2016/02/13/rails-5-does-not-halt-callback-chain-when-false-is-returned.html) - update test of
this behavior in callbacks_spec

Finished in 5.07 seconds
653 examples, 0 failures, 16 pending

britth commented:
since you probably have the most historical context given the 5.0 update, tagging you as well @dtykocki, if you happen to have time before leaving 😢

jeremywrowe reviewed (commented):
Looks good to me, but I'll defer to @dtykocki

curiousepic commented:
Welp, he got away. How do we feel about it?

dtykocki commented:
> Welp, he got away. How do we feel about it?

"I'm not dead yet"

This fork of Ripple is open-source FWIW.. the code changes look good to me but I would double check and run the integration tests just to be to be safe.

ptere commented on ripple.gemspec:
```diff
@@ -17,8 +17,8 @@ Gem::Specification.new do |gem|
   gem.add_dependency "riak-client", "~> 2.6"
-  gem.add_dependency "activesupport", "~> 5.0.7"
-  gem.add_dependency "activemodel", "~> 5.0.7"
+  gem.add_dependency "activesupport", "<= 5.1.7"
```
Why the constraint operator switch here?

jeremywrowe replied:
So that we can support older versions of rails. We should probably do >= 5.0.x also.

britth replied:
actually...after doing some more testing, the changes required for callback propagation won't work in versions before 5.1 so either we need to 
1) add in logic to support both in the `propagate_callbacks_to_embedded_associations` method, or
2) lock to a 5.1.x version

for now, I switched the operator back, but if we think supporting both is a better option, I can update that

britth committed:
Support for Rails 5.1

* ActiveSupport::Callbacks::Filters::Environment no longer supports
run_block; remove in propagate_callbacks_to_embedded_associations
* ActiveSupport::Callbacks::CallbackSequence call method no longer
supported; update logic in propagate_callbacks_to_embedded_associations
to call invoke_before on the sequence for :before callbacks and
invoke_after for :after callbacks
* Need to explicitly throw :abort to halt callback chain - update test of
this behavior in callbacks_spec

Finished in 5.07 seconds
653 examples, 0 failures, 16 pending
